tick/eval/node_types.go

```diff
@@ -0,0 +1,73 @@
+package eval
+
+import "github.com/influxdata/kapacitor/tick"
+
+/*
+	This file is patch!
+
+	We must extend the Node structs to implement NodeValueAccessor (see eval/node_value_accessors.go) and because
+	go permits defining new methods on non-local type, we create local types for them to use in eval package ONLY!
+*/
+
+type EvalBoolNode struct {
+	Node *tick.BoolNode
+}
+
+type EvalNumberNode struct {
+	Node *tick.NumberNode
+}
+
+type EvalRegexNode struct {
+	Node *tick.RegexNode
+}
+
+type EvalStringNode struct {
+	Node *tick.StringNode
+}
+
+type EvalBinaryNode struct {
+	Node *tick.BinaryNode
+}
+
+type EvalReferenceNode struct {
+	Node *tick.ReferenceNode
+}
+
+type EvalFunctionNode struct {
+	Node *tick.FunctionNode
+}
+
+type EvalUnaryNode struct {
+	Node *tick.UnaryNode
+}
+
+func createNodeValueAccessor(n tick.Node) NodeValueAccessor {
+	switch node := n.(type) {
+
+	case *tick.BoolNode:
+		return NodeValueAccessor(&EvalBoolNode{Node: node})
+
+	case *tick.NumberNode:
+		return NodeValueAccessor(&EvalNumberNode{Node: node})
+
+	case *tick.StringNode:
+		return NodeValueAccessor(&EvalStringNode{Node: node})
+
+	case *tick.RegexNode:
+		return NodeValueAccessor(&EvalRegexNode{Node: node})
+
+	case *tick.BinaryNode:
+		return NodeValueAccessor(&EvalBinaryNode{Node: node})
+
+	case *tick.ReferenceNode:
+		return NodeValueAccessor(&EvalReferenceNode{Node: node})
+
+	case *tick.FunctionNode:
+		return NodeValueAccessor(&EvalFunctionNode{Node: node})
+
+	case *tick.UnaryNode:
+		return NodeValueAccessor(&EvalUnaryNode{Node: node})
```

Comment:
By the way - this is great proposal! We can now change Eval{X}Node to contain data like expression cache or another info, but this for later change 👍